src/meta/size.rs

```diff
@@ -16,16 +16,16 @@ pub struct Size {
     unit: Unit,
 }
 
-impl<'a> From<&'a Metadata> for Size {
-    fn from(meta: &Metadata) -> Self {
-        let len = meta.len();
+impl Size {
+    pub fn from_bytes(meta: &Metadata) -> Option<Self> {
+        if !meta.is_file() {
+            return None;
+        }
 
-        Size::from_bytes(len as i64)
+        Some(Size::get_file_size(meta.len() as i64))
     }
-}
 
-impl Size {
-    fn from_bytes(len: i64) -> Self {
+    fn get_file_size(len: i64) -> Self {
```

Comment:
Why do you change the `from_bytes` into `get_file_size`? In my understanding the convention when you initialize a struct based on something the conventional name is :  `from_{the thing}`

Reply:
I did this to make the testing working, there still was `from_bytes`.
I separated the function which gets the file size separate to test it out otherwise I had to mock the `fs::Metadata` pass to `from_bytes`.

In any case the latest commit fixes this.